### PR TITLE
Rewrite player skill handling

### PR DESCRIPTION
**Pull request by Appled**
**Description**
At the moment if player learns new skill or updates an existing one, all player skill fields are looped through and each of them is updated. This generates unnecessary skill gain or skill updated messages for the player. Also, updating all skill fields makes some skills act strangely.

I have been rewriting this. Now if player learns new skill, server finds an available skill field slot for the new skill and saves its position to PlayerSkill. Since the field position is saved it allows skill update to update only the needed skill field without looping through all player skill fields.

In addition to skill handling:
Skills: Fix plate, mail, dual wield etc not working without relogging
Trainers: Fix trainers not showing all spells and skills
Transporters: Fix crash in passenger unload
Misc: Move some spell stuff from ObjectMgr to SpellMgr

As of writing this wotlk seems to be working as it should but cata is still untouched. I will finish this PR in the following days.
~~Note; this PR will **not** fix skill bonus values. I will leave that to another PR as this already is getting pretty large 😉~~

Personal todo list:
- [x] Rewrite wotlk skill handling
- [x] Rewrite cata skill handling
- [x] Cleanup skill related stuff
- [x] Move weapon and armor skills from playercreate spells to skills (fixes the skill gain messages in first login)
- [x] Weapon skills should be maxed all the time in Cata
- [x] Fix skill bonus fields
- [x] Add world_db_version rows to SQL files

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE.
- [x] Server startup.
- [x] Log into world.

***Multiversion Ingame Tests Performed:***
- [x] BC
- [x] WotLK
- [x] Cata

### CODE REVIEW

File: src/world/Spell/SpellMgr.cpp [hunk 1]
```diff
@@ -1330,7 +1391,138 @@ void SpellMgr::loadSpellAreas()
...
+
+    //                                                   0         1
+    const auto result = WorldDatabase.Query("SELECT spell_id, req_spell FROM spell_required");
+    if (result == nullptr)
+    {
+        sLogger.debug("SpellMgr : Loaded 0 spell required records. DB table `spell_required` is empty.");
+        return;
+    }
+
+    uint32 count = 0;
```
Comment (aaron02) [inline src/world/Spell/SpellMgr.cpp:1412]: _t

**Comment by Appled** [comment]
It's ready 😃

**Comment by schnek** [comment]
during a language problem "you don't know this language"
gm commands .speed copper

**Comment by Appled** [comment]
Old or new character?

**Comment by schnek** [comment]
old norm
new bug

**Comment by schnek** [comment]
> wotlk: gm commands not working during a language problem "you don't know this language" gm commands .speed copper

fix

**Pull request merged**